server/api/crud/model_monitoring/deployment.py

```diff
@@ -616,6 +636,234 @@ def _create_tsdb_tables(self):
+                )
+                tasks.append(task)
+
+        return mlrun.common.schemas.BackgroundTaskList(background_tasks=tasks)
+
+    def _get_monitoring_application_to_delete(
+        self,
+        delete_histogram_data_drift_app: bool = True,
+        delete_user_applications: bool = False,
+        user_application_list: list[str] = None,
+    ):
+        application_to_delete = []
+
+        if delete_user_applications:
+            if not user_application_list:
+                application_to_delete.extend(
+                    list(
+                        {
+                            app["metadata"]["name"]
+                            for app in self.list_model_monitoring_functions()
+                        }
+                    )
+                )
+            else:
+                for name in user_application_list:
+                    try:
+                        fn = server.api.crud.Functions().get_function(
+                            db_session=self.db_session,
+                            name=name,
+                            project=self.project,
+                        )
+                        if (
+                            fn["metadata"]["labels"].get(
+                                mm_constants.ModelMonitoringAppLabel.KEY
+                            )
+                            == mm_constants.ModelMonitoringAppLabel.VAL
+                        ):
+                            # checks if the given function is a model monitoring application
+                            application_to_delete.append(name)
+                        else:
+                            logger.warning(
+                                f"{name} is not a model monitoring application, skipping",
+                                project=self.project,
+                            )
+
+                    except mlrun.errors.MLRunNotFoundError:
+                        logger.warning(
+                            f"{name} is not found, skipping",
+                        )
+
+        if (
+            delete_histogram_data_drift_app
+            and mm_constants.HistogramDataDriftApplicationConstants.NAME
+            not in application_to_delete
+        ):
+            application_to_delete.append(
+                mm_constants.HistogramDataDriftApplicationConstants.NAME
+            )
+        return application_to_delete
+
+    @staticmethod
+    def _create_monitoring_function_deletion_background_task(
+        background_tasks: BackgroundTasks,
+        db_session: sqlalchemy.orm.Session,
+        project_name: str,
+        function_name: str,
+        auth_info: mlrun.common.schemas.AuthInfo,
+        delete_v3io_stream: bool,
+        access_key: str,
+    ):
+        background_task_name = str(uuid.uuid4())
+
+        # create the background task for function deletion
+        return server.api.utils.background_tasks.ProjectBackgroundTasksHandler().create_background_task(
```

Comment:
add `import server.api.utils.background_tasks` because otherwise it doesn't recognize the func